clients/drcachesim/tracer/output.cpp

```diff
@@ -1022,9 +1033,18 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
             }
             if (hit_window_end) {
                 if (op_offline.get_value() && op_split_windows.get_value()) {
-                    size_t add =
-                        instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
-                    buf_ptr += add;
+                    if (op_L0_filter_until_instrs.get_value() &&
+                        tracing_mode.load(std::memory_order_acquire) ==
+                            BBDUP_MODE_L0_FILTER) {
+                        NOTIFY(0, "Adding endpoint marker\n");
+                        size_t add = instru->append_marker(
+                            buf_ptr, TRACE_MARKER_TYPE_FILTER_ENDPOINT, 0);
+                        buf_ptr += add;
+                    } else {
+                        size_t add = instru->append_thread_exit(
```

Comment:
A thread exit is *also* needed isn't it?  The trace is malformed without it.

Reply:
Since we have a single file we don't have an exit after the filtered trace.

Reply:
This is the same as the prior one: I assume you do have separate files for -split_windows.

Reply:
This was another reason I thought you didn't have separate files, b/c it looked like you had removed the thread exit.  We do want the thread exit for separate files.